### DISCUSSION TITLE
Using multiple query parameters in an endpoint

### DISCUSSION THREAD

**Discussion by enes-kaplan** [ecyrbe/zodios#195] [Q&A] [answered]
Hello and thanks for this great library!

I have a question that I haven't been able to resolve, maybe there is a solution but I missed it in the docs and couldn't find an answer through Google/Stackoverflow either.

Let's say I have a GET endpoint that requires multiple query parameters in the request. How can I implement this with zodios?

```
{
  method: "get",
  path: "/users/:id", // id here is a single optional parameter
  alias: "getUser",
  description: "Get a user",
  response: z.object({
    id: z.number(),
    name: z.string(),
  })
}
```

But my endpoint is using multiple parameters like so: `/users?page=1&pageSize=10&name=test`

How can I add these parameters and type them as I want? I fiddled around and couldn't make it work with multiple parameters like `/users?page=:page&pageSize=:pageSize&name=:name`. I also want to type these input parameters.

**Comment by ecyrbe** [comment, marked as answer]
Hello,
Here is an example : https://github.com/ecyrbe/zodios/blob/main/examples/dev.to/articles.ts

**Reply by ecyrbe** [reply]
And here is how to call it: https://www.zodios.org/docs/client#zodiosget

**Reply by enes-kaplan** [reply]
That was quick, thanks for the reply. Sorry if I opened the issue in the wrong place.

This was one thing that I needed to resolve before using this library, I can define my API calls with zodios in my project now. I'll be sure to star the repo, hope that helps because this seems like a great library build on two other great libraries.

**Comment by ecyrbe** [comment]
Moved the issue to Q&A
